agent/engine/docker_task_engine.go

```diff
@@ -625,7 +654,19 @@ func (engine *DockerTaskEngine) startContainer(task *api.Task, container *api.Co
 			Error: CannotStartContainerError{fmt.Errorf("Container not recorded as created")},
 		}
 	}
-	return client.StartContainer(dockerContainer.DockerID, startContainerTimeout)
+	dockerContainerMD := client.StartContainer(dockerContainer.DockerID, startContainerTimeout)
+
+	// Get metadata through container inspection and available task information then write this to the metadata file
+	// Performs this in the background to avoid delaying container start
+	if dockerContainerMD.Error == nil && engine.cfg.ContainerMetadataEnabled == true {
```

Comment:
ahhhh I thought this was the docker metadata struct you implemented. Rename it to `metadataContainer`?